### ISSUE TITLE
Extend wrong thing to the import list automatically

### ISSUE THREAD

**Issue by berberman** [haskell/haskell-language-server#679]
I used ghcide master before HLS 0.7.0 released, and I encountered this many times. 
Importing things automatically is a cool feature, but here are some unsolved defects: https://github.com/haskell/ghcide/pull/930#issuecomment-739662283
BTW, before these problems get fixed, I would like to disable this automatic extending, but there seems to be no way :(

### Your environment
Output of `haskell-language-server --probe-tools` or `haskell-language-server-wrapper --probe-tools`:

```sh
haskell-language-server version: 0.7.0.0 (GHC: 8.10.2) (PATH: /home/berberman/.cabal/store/ghc-8.10.2/haskell-language-server-0.7.0.0-366830c6e84454f47d73ac01246969cb5c550194b34bde3428fbd57c6144f4d0/bin/haskell-language-server)
Tool versions found on the $PATH
cabal:		3.2.0.0
stack:		2.5.1
ghc:		8.10.2
```
Which lsp-client do you use:

vscode
Describe your project (alternative: link to the project):
a new draft project

### Steps to reproduce
![Peek](https://user-images.githubusercontent.com/26041945/102436244-15f87600-4053-11eb-9477-c9a4d7544073.gif)

### Expected behaviour
Don't add `return` to the import list.

### Actual behaviour
It was added incorrectly.

**Comment by berberman** [comment]
In this case, `buildable` was already in scope, imported by `import Distribution.Types.BuildInfo (BuildInfo (..))`:

![Peek](https://user-images.githubusercontent.com/26041945/102442099-8658c480-405e-11eb-9342-c20a926712a7.gif)

After I switched the order of `import Distribution.Types.BuildInfo (BuildInfo (..))` and `import Distribution.PackageDescription (repoLocation)`:

![Peek](https://user-images.githubusercontent.com/26041945/102442249-ed767900-405e-11eb-8cc8-9883c81d3278.gif)

🤦🏻‍♂️

**Comment by berberman** [comment]
I decided to switch back to HLS 0.6.0. Every time I press Enter after typed "BuildInfo", HLS will produce `import Distribution.PackageDescription (repoLocationBuildInfo, )` at [this line](https://github.com/berberman/arch-hs/blob/635c7453f08143a94b6063e3b71a0edc75be2b04/src/Distribution/ArchHs/Utils.hs#L42), which breaks typecheck, hvoer, diagnostic, etc. This is quite common, resulting in a bad user experience.

**Comment by berberman** [comment]
Maybe #670 could provide an option to disable `extendImportList` temporarily, before we elaborate this feature.

**Comment by jneira** [comment]
I would be nice to add a regression test and mark it as known broken, here and/or in ghcide.
I would keep open, as the plan is #685 disable the feature temporary, to check it is not reproduced when it is reenabled

**Comment by berberman** [comment]
Closed by #1266
